templates/server/src/main/resources/archetype-resources/core/src/main/java/__packageInPathFormat__/general/common/impl/security/BaseUserDetailsService.java

```diff
@@ -79,14 +80,64 @@ public UserDetails loadUserByUsername(String username) throws UsernameNotFoundEx
     return authorities;
   }
 
-  private Collection<String> getRoles(String username) {
+  /**
+   * Lists roles of AWS from IAM profile of pointed user.
+   *
+   * @param userName        given userName in AWS
+   * @param showErrorOutput flag marking outputing errors to console
+   * @return collection with user's roles
+   */
+  public static Collection<String> obtainRoles(String userName, boolean showErrorOutput) {
+    Collection<String> roles = new ArrayList<String>();
+
+    final String baseCommand = "aws iam list-groups-for-user --user-name";
+    String command = String.format("%s %s", baseCommand, userName);
+
+    BufferedReader reader = null;
+    Process process;
+    try {
+      process = Runtime.getRuntime().exec(command);
+      process.waitFor();
```

Comment:
so you have changed the app to connect to AWS. Nice feature.
But should we have that by default? What if project does not want to use AWS? What if "aws" command is not available?
Do not get me wrong: I love the idea, but this could be an option or maybe we need more like a documentation having that as  a code-snippet, or we could add some minimum module code for this that can be opted in.
See also:
https://github.com/oasp/oasp4j/issues/633